Cython/Compiler/Errors.py

```diff
@@ -284,3 +298,14 @@ def reset():
 
 def get_errors_count():
     return threadlocal.cython_errors_count
+
+def noexcept_performance_hint_helper(pos, function_name = None, void_return = False):
```

Comment:
1) "xyz_helper" is rarely a good name for a function since every function helps with something. That doesn't tell much about what it actually does. Actual verbs like `write_` or `issue_` seem better.
```suggestion
def write_noexcept_performance_hint(pos, function_name=None, void_return=False):
```
2) This seems too specific for `Errors.py`. You probably considered a couple of places before you put it here, and I agree that it's not easy to decide. But maybe somewhere in `PyrexTypes.py`? The hint that we give seems mostly related to C function types. I would consider adding a new module for this, but I'm not sure that we will have many cases where we issue the same hint from more than one place. We'll see about that…

Reply:
I created a new module (and also moved one function in ExprNodes that seemed to fit in the same place).

If you think that's excessive I can put it in PyrexTypes though

Reply:
Changed my mind because it caused a circular import and just moved it to PyrexTypes